share/functions/__fish_man_page.fish

```diff
@@ -1,4 +1,16 @@
 function __fish_man_page
-    man (basename (commandline -po; echo)[1]) ^/dev/null
-    or printf \a
+    # Get all commandline tokens not starting with "-"
+    set -l args (string join \n -- (commandline -po) | grep -v '^-')
```

Comment:
The `string join \n` was there because `grep` expects the tokens newline delimited, while `commandline -po` produces them space-delimited (maybe more precisely - it's that array-separator character). 

In any case - your suggestion is simpler - so I'm changing the code to that.